### PR TITLE
Adding round corners to UIView

### PR DESCRIPTION
**Pull request by Vic-L**
## Checklist

- [ ] New Extension
- [x] New Test
- [x] Changed more than one extension, but all changes are related
- [ ] Trivial change (doesn't require changelog)

**Gist**
- Overload `roundView()` with one that takes in parameters to add colored border to the view
- New method `nakedView()` to remove all masks of UIView

*PS its my first time contributing to a open source project. Thank you for your guidance 🙇 *

### CODE REVIEW

File: Sources/UIViewExtensions.swift [hunk 1]
```diff
@@ -565,9 +565,34 @@ extension UIView {
...
     public func roundView() {
-        self.layer.cornerRadius = min(self.frame.size.height, self.frame.size.width) / 2
+        self.setCornerRadius(radius: min(self.frame.size.height, self.frame.size.width) / 2)
+        self.layer.borderWidth = 0
+    }
+    
+    /**
+     EZSwiftExtensions - Mask square/rectangle UIView with a circular/capsule cover, with a border of desired color and width around it
+     
+     - Parameter withBorderColor: Color of circuler border
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: We dont need these comments. The args are seld documenting.

File: Sources/UIViewExtensions.swift [hunk 2]
```diff
@@ -565,9 +565,34 @@ extension UIView {
         self.layer.mask = mask
     }
 
-    /// EZSwiftExtensions
+    /**
+     EZSwiftExtensions - Mask square/rectangle UIView with a circular/capsule cover
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: We can put this comment on a single line.

File: Sources/UIViewExtensions.swift [hunk 3]
```diff
@@ -565,9 +565,34 @@ extension UIView {
...
+        self.layer.borderWidth = 0
+    }
+    
+    /**
+     EZSwiftExtensions - Mask square/rectangle UIView with a circular/capsule cover, with a border of desired color and width around it
+     
+     - Parameter withBorderColor: Color of circuler border
+     - Parameter withBorderWidth: Width of circuler border
+     */
+    public func roundView(withBorderColor color: UIColor, withBorderWidth width: CGFloat) {
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: I am thinking if color can be an optional parameter. That way roundView above can become a special case on the roundView with optional color and borderwidth = 0. Besides, a programmer can choose to have no color but some border width.

File: Sources/UIViewExtensions.swift [hunk 4]
```diff
@@ -565,9 +565,34 @@ extension UIView {
...
+        self.setCornerRadius(radius: min(self.frame.size.height, self.frame.size.width) / 2)
+        self.layer.masksToBounds = true
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: I am not aware what masks to bound does.

File: EZSwiftExtensionsTests/UIViewTests.swift [hunk 5]
```diff
@@ -60,5 +60,58 @@ class UIViewTests: XCTestCase {
...
+        tallView.roundView()
+        XCTAssertEqual(tallView.layer.cornerRadius, CGFloat(15.0))
+        XCTAssertEqual(tallView.layer.masksToBounds, true)
+    }
+    
+    func testRoundViewWithBorder() {
+        let view = UIView(x: 0, y: 0, w: 100, h: 100)
+        let color = UIColor.blue
+        let borderWidth: CGFloat = 5.0
+        view.layer.borderColor = nil // override default view.layer.borderColor
```
Comment (Khalian) [inline EZSwiftExtensionsTests/UIViewTests.swift]: I am not wholly comfortable with this nil set and then assert. Maybe consider optional bordercolor as arg or something. Then asset the content of the optional to find nothing there.

File: Sources/UIViewExtensions.swift [hunk 6]
```diff
@@ -565,9 +565,34 @@ extension UIView {
...
+        self.layer.borderWidth = width
+        self.layer.borderColor = color.cgColor
+    }
+    
+    /**
+     EZSwiftExtensions - Remove all masking around UIView
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: This comment can be in one line.

File: Sources/UIViewExtensions.swift [hunk 7]
```diff
@@ -565,9 +565,34 @@ extension UIView {
...
+     */
+    public func nakedView() {
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: Whats the use case for this?

**Comment by Khalian** [comment]
Could you also add a changelog entry and squash your commits together?

File: Sources/UIViewExtensions.swift (hunk 7 above)
Comment (Vic-L) [inline Sources/UIViewExtensions.swift]: Ah I used it for a rounded UIImageView showing a user's profile picture.

When user logs in and his/her profile picture is loaded, the mask needs to be on to make the image rounded, and with a border around it.
When user logs out, the image will change to a image asset, and the mask needs to be taken out.

I thought a helper method to remove any mask present would be helpful.

File: Sources/UIViewExtensions.swift (hunk 4 above)
Comment (Vic-L) [inline Sources/UIViewExtensions.swift]: I was under the impression that the mask will not resize along with the view is masksToBounds is false. Apparently I'm wrong :( I'll remove it

This is the code snippet that i used to experiment
```
override func viewDidLoad() {
        let view = UIView(frame: CGRect(x: 0, y: 0, w: 250, h: 250))
        view.roundView()
        view.backgroundColor = .green
        view.layer.masksToBounds = false // toggle here
        
        self.view.addSubview(view)
        if #available(iOS 10.0, *) {
            timer = Timer.scheduledTimer(withTimeInterval: 0.5, repeats: true, block: { (_timer) in
                view.snp.updateConstraints({ (make) in
                    make.height.equalTo(ez.screenHeight)
                    make.width.equalTo(ez.screenWidth)
                })
                
                self.view.setNeedsLayout()
                
                UIView.animate(withDuration: 2, animations: {
                    self.view.layoutIfNeeded()
                }, completion: { (completed) in
                    if completed {
                        view.snp.updateConstraints({ (make) in
                            make.height.equalTo(100)
                            make.width.equalTo(100)
                        })
                        
                        self.view.setNeedsLayout()
                    }
                    
                    UIView.animate(withDuration: 2, animations: {
                        self.view.layoutIfNeeded()
                    })
                })
            })
        } else {
            // Fallback on earlier versions
        }
        
    }
```

**Comment by Vic-L** [comment]
oops i think i did the wrong thing
`git push -f` ~= `squash` eh :(

File: EZSwiftExtensionsTests/UIViewTests.swift [hunk 8]
```diff
@@ -60,5 +60,56 @@ class UIViewTests: XCTestCase {
...
+        XCTAssertEqual(tallView.layer.borderWidth, 0)
+        XCTAssertNotEqual(tallView.layer.borderColor, UIColor.clear.cgColor)
+        
+        tallView.roundView()
+        XCTAssertEqual(tallView.layer.cornerRadius, CGFloat(15.0))
+        XCTAssertEqual(tallView.layer.borderWidth, 0)
+        XCTAssertEqual(tallView.layer.borderColor, UIColor.clear.cgColor)
+    }
+    
+    func testRoundViewWithBorder() {
```
Comment (Khalian) [inline EZSwiftExtensionsTests/UIViewTests.swift]: Minor : I think you can call this method testRoundViewWithBorderAndColor.

**Comment by Vic-L** [comment]
👌 
Many thx for your patience!

Great work on the pod btw!

**Pull request merged**